src/meta/KVJobManager.cpp

```diff
@@ -0,0 +1,647 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include <chrono>
+#include <string>
+#include <thread>
+#include <vector>
+#include <boost/stacktrace.hpp>
+#include <gtest/gtest.h>
+#include <folly/futures/Future.h>
+#include <folly/synchronization/Baton.h>
+#include <folly/String.h>
+
+#include "base/Base.h"
+#include "http/HttpClient.h"
+#include "kvstore/Common.h"
+#include "kvstore/KVIterator.h"
+#include "meta/processors/Common.h"
+#include "meta/TaskDescription.h"
+#include "meta/JobStatus.h"
+#include "meta/KVJobManager.h"
+#include "meta/MetaServiceUtils.h"
+#include "webservice/Common.h"
+
+using nebula::kvstore::ResultCode;
+using nebula::kvstore::KVIterator;
+using ErrOrString = nebula::ErrorOr<nebula::kvstore::ResultCode, std::string>;
+using ErrOrInt = nebula::ErrorOr<nebula::kvstore::ResultCode, int>;
+
+namespace nebula {
+namespace meta {
+
+KVJobManager::KVJobManager() {
+}
+
+KVJobManager* KVJobManager::getInstance() {
+    static KVJobManager inst;
+    return &inst;
+}
+
+bool KVJobManager::init(nebula::kvstore::KVStore* store,
+                        nebula::thread::GenericThreadPool *pool) {
+    if (!store || !pool) return false;
+    KVJobManager* inst = KVJobManager::getInstance();
+    inst->kvStore_ = store;
+    inst->pool_ = pool;
+
+    kCurrId = kJobPrefix + "__id";
+    kJobKey = kJobPrefix + "_";
+    kJobArchive = kJobPrefix + "archive_";
+
+    executor_ = std::thread(&KVJobManager::runBackground, this);
+    executor_.detach();
```

Comment:
why detach? the function runBackground use some global resource, and if we do not control the thread exit sequence, maybe will take some problem

Reply:
thx, good point, use thread::GenericWorker instead